pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -57,7 +58,7 @@ import (
 
 var (
 	edsClusterMutex sync.RWMutex
-	edsClusters     = map[string]*EdsCluster{}
+	edsClusters     = map[string]map[model.Locality]*EdsCluster{}
```

Comment:
This is another place where it doesn't seem safe ( in terms of perf ).

Given that there is no bench test or data - and is so late in 1.1, it would be good to make sure a default locality is used when the feature is disabled. 

It seems this change does have an impact even on people not using the feature - which is not ok.